### PR TITLE
[bitnami/thanos]Add feature to configure cache settings store gateway.

### PR DESCRIPTION
**Pull request by Whyeasy**
**Description of the change**

With this change the capability is added to configure the index caching within the Store Gateway. More info: https://thanos.io/components/store.md/#index-cache

**Benefits**

Have the option to move the index cache to a `memcached` deployment or to a `caching bucket`.

**Possible drawbacks**

When a existing configmap is provided, it should contain the data with the key `config.yml`. As is done within the generated configmap.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-version: 2.1.1
+version: 2.2.1
```
Comment (marcosbc) [inline bitnami/thanos/Chart.yaml:2]: I understand this would be in sync with #3387

File: bitnami/thanos/templates/storegateway/statefulset.yaml [hunk 2]
```diff
@@ -77,6 +77,9 @@ spec:
             - --http-address=0.0.0.0:10902
             - --data-dir=/data
             - --objstore.config-file=/conf/objstore.yml
+            {{- if or .Values.storegateway.config .Values.storegateway.existingConfigmap}}
```
Comment (marcosbc) [inline bitnami/thanos/templates/storegateway/statefulset.yaml:80]:
```suggestion
            {{- if or .Values.storegateway.config .Values.storegateway.existingConfigmap }}
```

File: bitnami/thanos/templates/storegateway/statefulset.yaml [hunk 3]
```diff
@@ -105,13 +108,22 @@ spec:
               {{- end }}
             - name: data
               mountPath: /data
+            {{- if or .Values.storegateway.config .Values.storegateway.existingConfigmap}}
```
Comment (marcosbc) [inline bitnami/thanos/templates/storegateway/statefulset.yaml:111]:
```suggestion
            {{- if or .Values.storegateway.config .Values.storegateway.existingConfigmap }}
```

File: bitnami/thanos/Chart.yaml (hunk 1 above)
Comment (marcosbc) [inline bitnami/thanos/Chart.yaml:2]: Could you merge from master and fix the merge conflicts?

File: bitnami/thanos/templates/storegateway/statefulset.yaml [hunk 4]
```diff
@@ -105,13 +108,22 @@ spec:
...
+              mountPath: /conf/cache
+            {{- end}}
       volumes:
         - name: objstore-config
           secret:
             secretName: {{ include "thanos.objstoreSecretName" . }}
             {{- if .Values.existingObjstoreSecretItems }}
             items: {{- toYaml .Values.existingObjstoreSecretItems | nindent 14 }}
             {{- end }}
+        {{- if or .Values.storegateway.config .Values.storegateway.existingConfigmap}}
```
Comment (marcosbc) [inline bitnami/thanos/templates/storegateway/statefulset.yaml:122]:
```suggestion
        {{- if or .Values.storegateway.config .Values.storegateway.existingConfigmap }}
```

**Comment by Whyeasy** [comment]
Rebase went wrong, closing this PR to fix changed files.

**Pull request closed without merging**